homeworks/03-data-structure/binary-tree/binaryTree.ts

```diff
@@ -0,0 +1,208 @@
+export const enum TraverseType {
+    PreOrder = 'Preorder',
+    InOrder = 'InOrder',
+    PostOrder = 'PostOrder',
+    BreadthFirst = 'BreadthFirst'
+}
+
+type TraverseFilterer<ValueType> = (node?: BinaryTreeNode<ValueType>) => boolean
+
+class BinaryTreeNode<ValueType> {
+    public value: ValueType
+    public parent: BinaryTreeNode<ValueType>
+    public leftChild: BinaryTreeNode<ValueType>
+    public rightChild: BinaryTreeNode<ValueType>
+    public columnNumber: number
+
+    constructor(newValue: ValueType, newParent: BinaryTreeNode<ValueType> = null, newColumnNumber: number = 0) {
+        this.value = newValue
+        this.parent = newParent
+        this.leftChild = this.rightChild = null
+        this.columnNumber = newColumnNumber
+    }
+}
```

Comment:
`TreeNode` doesn't correspond to the required interface at all. Should be only `value`, `left`, and `right` fields.
Also, no needs to do it as a separate class. It can be just an interface.